lib/internal/fs/streams.js

```diff
@@ -386,22 +386,72 @@ WriteStream.prototype.open = openWriteFs;
 
 WriteStream.prototype._construct = _construct;
 
+function writeAll(data, pos, cb, retries = 0) {
+  const len = data.length;
+  this[kFs].write(this.fd, data, 0, len, pos, (er, bytesWritten) => {
+    if (er?.code === 'EAGAIN') {
+      er = null
+      bytesWritten = 0
+    }
+
+    if (this.destroyed || er) {
+      return cb(er);
+    }
+
+    retries = bytesWritten ? 0 : retries + 1
+
+    if (retries > 5) {
+      return cb(new Error('writev failed'));
```

Comment:
What are we trying for 5 times here? Writing a non-zero number of bytes?

Reply:
Yes

Reply:
Maybe add a comment here.